packages/core/src/modules/proofs/protocol/v1/handlers/V1RequestPresentationHandler.ts

```diff
@@ -0,0 +1,120 @@
+import type { Handler, HandlerInboundMessage } from '../../../../../agent/Handler'
+import type { DidCommMessageRepository } from '../../../../../storage/didcomm/DidCommMessageRepository'
+import type { MediationRecipientService } from '../../../../routing'
+import type { ProofResponseCoordinator } from '../../../ProofResponseCoordinator'
+import type { ProofRecord } from '../../../repository/ProofRecord'
+import type { V1ProofService } from '../V1ProofService'
+
+import { createOutboundMessage, createOutboundServiceMessage } from '../../../../../agent/helpers'
+import { ServiceDecorator } from '../../../../../decorators/service/ServiceDecorator'
+import { DidCommMessageRole } from '../../../../../storage'
+import { ProofProtocolVersion } from '../../../models/ProofProtocolVersion'
+import { V1RequestPresentationMessage } from '../messages'
+
+export class V1RequestPresentationHandler implements Handler {
+  private proofService: V1ProofService
+  private agentConfig: AgentConfig
+  private proofResponseCoordinator: ProofResponseCoordinator
+  private mediationRecipientService: MediationRecipientService
+  private didCommMessageRepository: DidCommMessageRepository
+  public supportedMessages = [V1RequestPresentationMessage]
+
+  public constructor(
+    proofService: V1ProofService,
+    agentConfig: AgentConfig,
+    proofResponseCoordinator: ProofResponseCoordinator,
+    mediationRecipientService: MediationRecipientService,
+    didCommMessageRepository: DidCommMessageRepository
+  ) {
+    this.proofService = proofService
+    this.agentConfig = agentConfig
+    this.proofResponseCoordinator = proofResponseCoordinator
+    this.mediationRecipientService = mediationRecipientService
+    this.didCommMessageRepository = didCommMessageRepository
+  }
+
+  public async handle(messageContext: HandlerInboundMessage<V1RequestPresentationHandler>) {
+    const proofRecord = await this.proofService.processRequest(messageContext)
+    if (this.proofResponseCoordinator.shouldAutoRespondToRequest(proofRecord)) {
+      return await this.createPresentation(proofRecord, messageContext)
+    }
+  }
+
+  private async createPresentation(
+    record: ProofRecord,
+    messageContext: HandlerInboundMessage<V1RequestPresentationHandler>
+  ) {
+    const requestMessage = await this.didCommMessageRepository.getAgentMessage({
+      associatedRecordId: record.id,
+      messageClass: V1RequestPresentationMessage,
+    })
+
+    const indyProofRequest = requestMessage.indyProofRequest
+
+    this.agentConfig.logger.info(
+      `Automatically sending presentation with autoAccept on ${this.agentConfig.autoAcceptProofs}`
+    )
+
+    if (!indyProofRequest) {
+      this.agentConfig.logger.error('Proof request is undefined.')
+      return
+    }
+
+    const retrievedCredentials = await this.proofService.getRequestedCredentialsForProofRequest({
+      proofRecord: record,
+      config: {
+        filterByPresentationPreview: false,
```

Comment:
I think we should be passing true here
```suggestion
        filterByPresentationPreview: true,
```